### PR TITLE
Sample attachments

### PR DESCRIPTION
**Pull request by ericenns**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds in support for attaching files to Samples. Currently this only allows a User with `role >= Maintainer` to attach files to a Sample.

This PR also adds an id to the layout component div for flashes so that `turbo_stream` requests can display flashes. This is used in the `create` and `destroy` actions.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

image

image

image

image

image

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Migrate your DB with `bin/rails db:migrate`
2. Navigate to a Sample and click `Upload File`
3. Select a file and click `Upload File`
4. Confirm that it appears in the table
5. Repeat this process and confirm that a new file appears in the table and the existing file remains.
6. Click "Delete" on one of the files you uploaded, verify that the confirm dialog appears, and verify that it is removed from the table after clicking confirm.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

**Comment by joshsadam** [comment]
The Upload New File modal could use some text.  Even if it just says "You can upload any file type you want", or what happens  after upload.

File: test/models/attachment_test.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class AttachmentTest < ActiveSupport::TestCase
+  # test "the truth" do
```
Comment (joshsadam) [inline test/models/attachment_test.rb:6]: **nitpick:** Can you remove this empty test or fill it in?

**Review by joshsadam — CHANGES_REQUESTED** [review]
This looks really good.  A couple nitpick comments, but I think overall great work.

**Comment by joshsadam** [comment]
We need an empty state for when there are no files.  I created an issue for it to be handled in the future: #152

File: test/models/attachment_test.rb (hunk 1 above)
Comment (ericenns) [reply test/models/attachment_test.rb]: Addressed in 009ade8a

File: app/views/projects/samples/attachments/_attachment.html.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  <th scope="row" class="px-6 py-4"><%= link_to attachment.file.filename,
+    download_namespace_project_sample_attachment_path(
+      sample_id: @sample.id,
+      id: attachment.id
+    ),
+    data: {
+      turbo: false
+    } %></th>
+  <td class="px-6 py-4"><%= number_to_human_size(attachment.file.blob.byte_size) %></td>
+  <td class="px-6 py-4"><%= l(attachment.created_at) %></td>
```
[the remark this answers is not in the document]
Comment (ericenns) [reply app/views/projects/samples/attachments/_attachment.html.erb]: Addressed in ff635e8

**Comment by ericenns** [comment]
> The Upload New File modal could use some text.  Even if it just says "You can upload any file type you want", or what happens  after upload.

Let's hold off on this for now. This upload modal is the first iteration which just allows single file uploads, in the future this will support uploading pair end data that needs to be related to each other via the `metadata` params. As such this can be left for later as this is expected to undergo significant changes.

**Review by deepsidhu85 — APPROVED** [review]
This looks good to me and working as expected! Great work

File: test/models/attachment_test.rb (hunk 1 above)
Comment (joshsadam) [reply test/models/attachment_test.rb]: Thanks

**Review by joshsadam — APPROVED** [review]
Great stuff... again :)

**Pull request merged**